### PR TITLE
[CHI-76] Add tooltip component

### PR DESCRIPTION
**Pull request by dani-cl-madrid**
JavaScript component for tooltips. Fully integrated with Popper, doesn't work without it.

### CODE REVIEW

File: src/website/views/javascript/tooltip.pug [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    <script>chi.tooltip(document.getElementById('tootltip-1'));</script>
+
+h3 Positioning
+p.-text
+  | By default, tooltip's position on top of an element. To alter position, use
+  | the <code>data-position</code> attribute. Valid values are <code>top</code>,
+  | <code>right</code>, <code>bottom</code>, <code>left</code>. You can pass an array
+  | of Elements and initialize all at once.
+.example.-mb--3
+  .-p--3.-d--flex
```
Comment (jllr) [inline src/website/views/javascript/tooltip.pug]: @dani-cl-madrid remove .-d--flex class

Comment (jllr) [inline src/website/views/javascript/tooltip.pug]:
```suggestion
  .-p--3
```

File: src/website/views/javascript/tooltip.pug [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    button(data-tooltip='Your top tooltip text').a-btn.-mr--2 Top Tooltip
```
Comment (jllr) [inline src/website/views/javascript/tooltip.pug]: @dani-cl-madrid add margin bottom when smaller sizes and remove when goes medium

File: src/chi/javascript/tooltip.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      self._preAnimationTransformStyle = null;
+      self._postAnimationTransformStyle = data.styles.transform;
+      if (data.placement.indexOf("top") === 0) {
+        self._preAnimationTransformStyle = `translate3d(${data.popper.left}px, ${data.popper.top + 20}px, 0px)`;
+      } else if (data.placement.indexOf("right") === 0) {
+        self._preAnimationTransformStyle = `translate3d(${data.popper.left + 20}px, ${data.popper.top}px, 0px)`;
+      } else if (data.placement.indexOf("bottom") === 0) {
+        self._preAnimationTransformStyle = `translate3d(${data.popper.left}px, ${data.popper.top - 20}px, 0px)`;
+      } else if (data.placement.indexOf("left") === 0) {
+        self._preAnimationTransformStyle = `translate3d(${data.popper.left - 20}px, ${data.popper.top+20}px, 0px)`;
```
Comment (mattnickles) [inline src/chi/javascript/tooltip.js]: Left and right have different top values. Can we update the left to render the same as the right?

File: src/website/views/javascript/tooltip.pug [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+h3 Base
+p.-text
+  | To render a tooltip, apply the <code>data-tooltip</code> attribute to an element. Then,
+  | you can initialize it with the chi.tooltip factory method.
+.example.-mb--3
+  .-p--3
+    button(data-tooltip='Your tooltip text on a button').a-btn.-mr--2 Tooltip
+  :code(lang='html')
+    <button id='tootltip-1' class="a-btn" data-tooltip="Your tooltip text on a button">Tooltip</button>
```
Comment (mattnickles) [inline src/website/views/javascript/tooltip.pug]: change all instances of "tootltip" to "tooltip"

File: src/chi/javascript/tooltip.js (hunk 3 above)
Comment (mattnickles) [reply src/chi/javascript/tooltip.js]: To clarify, let's remove the +20 from left

**Pull request merged**